components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/AuthenticationContext.java

```diff
@@ -59,6 +62,16 @@ public void setUser(User user) {
         this.user = user;
     }
 
+    public Object getProperty(String key) {
+
+        return properties.get(key);
+    }
+
+    public void setProperty(String key, Object value) {
```

Comment:
let's rename to addProperty

e.g. https://github.com/wso2/carbon-identity-framework/blob/fa2db1277cc0a65c30a6b1c96cf2a683a9917fda/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticationResult.java#L104